lib/ansible/plugins/connection/docker.py

```diff
@@ -132,8 +132,8 @@ def _get_docker_version(self):
 
         cmd, cmd_output, err, returncode = self._old_docker_version()
         if returncode == 0:
-            for line in cmd_output.split('\n'):
-                if line.startswith('Server version:'):  # old docker versions
+            for line in cmd_output.split(b'\n'):
+                if line.startswith(b'Server version:'):  # old docker versions
```

Comment:
The reason it seems better to convert to text is that the strategy Ansible is taking is to convert at incoming borders from bytes to text and convert at outgoing borders from text to bytes.  We disregard this in a few places (for instance, the connection plugins' API returning bytes).  The difference between those example, however, is that we do not know how calling code will choose to use the strings returned by the connection API.  Therefore we can't know whether it's most appropriate to sacrifice information (via errors='replace') or fail (via errors='strict') if there is a problem decoding the string.  In this case, we know exactly how the string is going to be used in the end.  It needs to be used as a text string.  So that means that we should be converting the string into a text string at the border where it enters Ansible and mixing it only with other text strings throughout its path to its destination.